includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -1127,7 +1132,7 @@ public function ajax_add_to_cart() {
 			WC()->cart->add_to_cart( $product->get_id(), $qty, $variation_id, $attributes );
 		}
 
-		if ( 'simple' === $product_type || 'subscription' === $product_type ) {
+		if ( 'simple' === $product_type || in_array( $product_type, $subscription_types, true ) ) {
```

Comment:
I took this approach from the function `get_product_price()` on this same file. [Line 222](https://github.com/Automattic/woocommerce-payments/blob/f24aed493ee7f6d50b5caa71b060232a7b22be79/includes/class-wc-payments-payment-request-button-handler.php#L222). So maybe `$subscription_types` it could be abstracted as a class static variable and accessed in both places? I wanted to keep it as simple as possible, but let me know and I can make the change.